cloud/metadata/bitbucket.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+//nolint:revive
+package metadata
+
+import "github.com/terramate-io/terramate/cmd/terramate/cli/bitbucket"
+
+type BitbucketPullRequest struct {
+	Author *BitbucketUser `json:"author,omitempty"`
+
+	// Participants contains reviewers, users that approved (assigned or not assigned), and commenters.
+	Participants []BitbucketActor `json:"participants,omitempty"`
+
+	// Reviewers contains only assigned reviewers.
+	Reviewers []BitbucketUser `json:"reviewers,omitempty"`
+}
+
+type BitbucketUser struct {
+	UUID        string `json:"uuid,omitempty"`
+	DisplayName string `json:"display_name,omitempty"`
+	AccountID   string `json:"account_id,omitempty"`
+	Nickname    string `json:"nickname,omitempty"`
+	AvatarURL   string `json:"avatar_url,omitempty"`
+}
+
+type BitbucketActor struct {
+	Type string         `json:"type,omitempty"`
+	User *BitbucketUser `json:"user,omitempty"`
+	// Role can be PARTICIPANT, REVIEWER.
+	Role     string `json:"role,omitempty"`
+	Approved *bool  `json:"approved,omitempty"`
+	// State can be changes_requested, approved.
+	State          string `json:"state,omitempty"`
+	ParticipatedOn string `json:"participated_on,omitempty"`
+}
+
+func NewBitbucketPullRequest(in *bitbucket.PR) (*BitbucketPullRequest, error) {
+	author, err := NewBitbucketUser(&in.Author)
+	if err != nil {
+		return nil, err
+	}
+
+	participants := make([]BitbucketActor, 0, len(in.Participants))
+	for _, e := range in.Participants {
+		p, err := NewBitbucketActor(&e)
+		if err != nil {
+			return nil, err
+		}
+		participants = append(participants, *p)
+	}
+
+	reviewers := make([]BitbucketUser, 0, len(in.Reviewers))
+	for _, e := range in.Reviewers {
+		p, err := NewBitbucketUser(&e)
+		if err != nil {
+			return nil, err
+		}
+		reviewers = append(reviewers, *p)
+	}
+
+	return &BitbucketPullRequest{
+		Author:       author,
+		Participants: participants,
+		Reviewers:    reviewers,
+	}, nil
+}
+
+func NewBitbucketUser(in *bitbucket.User) (*BitbucketUser, error) {
```

Comment:
Never a non-nil error is returned so it'd better to remove it from the signature:
```suggestion
func NewBitbucketUser(in *bitbucket.User) *BitbucketUser {
```